tofhir-engine/src/main/scala/io/tofhir/engine/data/read/FileDataSourceReader.scala

```diff
@@ -35,15 +35,21 @@ class FileDataSourceReader(spark: SparkSession) extends BaseDataSourceReader[Fil
     var rowIndex: Integer = 0
     //Based on source type
     val resultDf = mappingSource.sourceType match {
-        case SourceFileFormats.CSV =>
+        case SourceFileFormats.CSV | SourceFileFormats.TSV =>
           //Options that we infer for csv
           val inferSchema = schema.isEmpty || mappingSource.preprocessSql.isDefined
           val csvSchema = if(mappingSource.preprocessSql.isDefined) None else schema
           //val enforceSchema = schema.isDefined
           val includeHeader = mappingSource.options.get("header").forall(_ == "true")
-          val fileFilter = mappingSource.options.getOrElse("pathGlobFilter", "*.csv")
-          //Other options
-          val otherOptions = mappingSource.options.filterNot(o => o._1 == "header" || o._1 == "inferSchema" || o._1 == "enforceSchema")
+          val fileFilter = mappingSource.options.getOrElse("pathGlobFilter", s"*.{${SourceFileFormats.CSV},${SourceFileFormats.TSV}}")
```

Comment:
This filter might lead some unexpected results. For example, I have a directory which has two CSV and two TSV files. I want to read only TSV files but it will read all of them because of this filter, right ?

Reply:
I thought the similar thing but we are setting data source with its extension (patients.csv). So it should not be problem for batch jobs. Regarding streaming ones, could be a problem. I think it's better to refactor here.